### ISSUE TITLE
dramatically long time to show output in file function  (or when apply)

### ISSUE THREAD

**Issue by dzirg44** [hashicorp/terraform#27743]
### Terraform Version

```
Terraform v0.14.5
```

### Debug Output
just long time to wait

### Expected Behavior
quick show big file

### Actual Behavior
1.6 mb file takes 5 minutes

### Steps to Reproduce

1. make big file or for example you can use `https://github.com/jetstack/cert-manager/releases/download/v1.1.0/cert-manager.crds.yaml`
2. read this file via `file` function `file(cert-manager.crds.yaml)`
3. create output
4. apply

### Additional Context
in case if it is not easy to improve speed, maybe it will be a nice idea to create an option or env variable , like
TF_BIG_FILES=false
where we can see something like
```
output = 1 
aaaa
bbbb
ccccc
....... showing big resources is disabled due to performance , but you can enable it via ENV variable 
```

**Comment by alisdair** [comment]
Hi @dzirg44, thanks for reporting this. I'm not sure what's happening here exactly, but I can confirm that this specific file does seem to cause a pathological performance problem.

It's not as simple as loading any large file, as you can see from this attempted reproduction:

1. Create a 2MB text file: `base64 < /dev/random | dd of=bigfile bs=1024 count=2048`
2. Use this configuration:

    ```hcl
    output "foo" {
      value = file("bigfile")
    }
    ```
3. Run `terraform apply`, and see it execute and render the output reasonably quickly (within a few seconds)

Even more confusing, if you finish this apply, then change the `file()` call to point to your supplied YAML file, Terraform again executes quickly. It seems to only cause a problem when creating a _new_ output using this file contents.

There must be something about the particular file you linked to which is causing the issue, which is confusing. To understand what's going on here I think we'll need to run a profile on Terraform in this specific case.

In case someone else picks this up, my successful reproduction steps are:

1. Download the linked YAML file as "creds.yaml"
2. Use this configuration:

    ```hcl
    output "foo" {
      file("creds.yaml")
    }
    ```
3. With no state file, run `terraform apply`
4. See Terraform hang for several minutes after displaying the resource change summary

**Comment by alisdair** [comment]
Another note: it does not appear to be the `file` call that is the issue here, as an output with `value = substr(file("creds.yaml"), 0, 1000)` executes quickly. I guess it might be the output rendering code.

**Comment by alisdair** [comment]
There's a proposed fix for this issue in #27746. I would note that this is a very unusual edge case that you appear to have hit here. Is a 25,000 line string output a real use case for you, or was this just something you discovered while working on something else?

**Comment by dzirg44** [comment]
yes, especially when I work with long crds . at first I thought that the problem was in community provider which I use for manifests. But after this I dicovered, that even output takes a lot of time.
I made a quick workaround, use `split` function 
 ```hcl
splited_yaml = split("---", file("${path.module}/files/${var.crd_version}/cert-manager.crds.yaml"))
```
it runs quicker than just plan file.
I think this fix can be very helpful for new kubernetes provider, which operates crds.
Thank you for your help!
